pool/hub.go

```diff
@@ -623,14 +631,24 @@ func (h *Hub) shutdown() {
 
 // Run handles the process lifecycles of the pool hub.
 func (h *Hub) Run(ctx context.Context) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	var wg sync.WaitGroup
 	wg.Add(3)
 	go func() {
 		h.endpoint.run(ctx)
 		wg.Done()
 	}()
 	go func() {
-		h.chainState.handleChainUpdates(ctx)
+		err := h.chainState.handleChainUpdates(ctx)
+		if err != nil {
+			// Ensure the context is canceled so the remaining goroutines exit
+			// when there was an error that caused the chain update handler to
+			// exit prematurely.
+			log.Error(err)
```

Comment:
Maybe skip this log if `errors.Is(err, context.Canceled)`?

Reply:
`handleChainUpdates` doesn't return `ctx.Err()` since it's expected that the context will be closed.  Since it doesn't return that, it can never be `context.Canceled`.

Reply:
That is true if `handleChainUpdates` returns via the `case <-ctx.Done():` path, but it might return via another path.

E.g. if the ctx is canceled during the call to `cs.cfg.GetBlock(ctx, &header.PrevBlock)`, that func might return `context.Canceled`, which in turn will get returned out of `handleChainUpdates`.